lib/galaxy/tools/deps/resolvers/conda.py

```diff
@@ -153,7 +153,7 @@ def resolve_all(self, requirements, **kwds):
                 dependency = MergedCondaDependency(
                     self.conda_context,
                     self.conda_context.env_path(env),
-                    exact=self.versionless or requirement.version is None,
+                    exact=not self.versionless or requirement.version is None,
```

Comment:
If ``requirement.version`` is ``None``, this ``Dependency`` object is considered ``exact`` I believe (it might be better to think of ``exact`` as "not inexact" or "not an approximation). So I don't think your expression is correct.